### PR TITLE
Tutorial: llvm2cpg

### PR DESCRIPTION
**Pull request by khorne3**

### CODE REVIEW

File: using-ocular/tutorials/cpg-llvm.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+```llvm
+; main.ll
+declare void @callee(i32 %sink)
+define void @caller(i32 %x) {
+    call void @callee(i32 %x)
+  ret void
+}
+```
+
+Run **llvm2cpg against your newly-created file:
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
Run **llvm2cpg** against your newly-created file:
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+clang -c -emit-llvm main.c -o main.bc
+```
+
+Alternatively, you can run the same command, but with LTO in **Full** mode:
+
+```bash
+clang -c -flto main.c -o main.o
+```
+
+Regardless of whether your output is **main.bc** or **main.c**, both contain bitcode. You can verify as follows:
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
Regardless of whether your output is **main.bc** or **main.o**, both contain bitcode. You can verify as follows:
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+```bash
+> file main.o main.bc
+main.o:  LLVM bitcode, wrapper x86_64
+main.bc: LLVM bitcode, wrapper x86_64
+```
+
+Both types of files can be passed to llvm2cpg (e.g., `llvm2cpg main.bc` or `llvm2cpg main.c`).
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
Both types of files can be passed to llvm2cpg (e.g., `llvm2cpg main.bc` or `llvm2cpg main.o`).
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+Using embedded bitcode is ideal, since it results in the most straightforward integration and can be added to an existing build system without affecting the resulting software.
+
+### Obtaining Bitcode for Your Source Code
+
+Getting bitcode for your projects can be less straightforward than our projects, especially given the various build systems in use. One thing to remember is that you'll need to inject one of the following flags into your build system:
+
+| **Flag** | **Notes** |
+| - | - |
+| `-emit-llvm` | The build doesn't finish (and linking fails since no object files are produced), but all bitcode files are available |
+| `flto` | The build completes, and all of the created intermediate object files contain bitcode |
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
| `-flto` | The build completes, and all of the created intermediate object files contain bitcode |
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+| `fembed-bitcode` | The build completes and the resulting binary contains bitcode |
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
| `-fembed-bitcode` | The build completes and the resulting binary contains bitcode |
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+```bash
+xcodebuild OTHER_CFLAGS=-fembed-bitcode OTHER_CPLUSPLUSFLAGS=-fembed-bitcode OTHER_LDFLAGS=-fembed-bitcode
+```
+
+We recommend that users of other build systems look into using [whole-program-llvm](https://github.com/travitch/whole-program-llvm).
+
+#### Caveats
+
+* The `-fembed-bitcode` flag may not work on macOS if a project links to a static library that wasn'tt compiled with embedded bitcode support
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
* The `-fembed-bitcode` flag may not work on macOS if a project links to a static library that wasn't compiled with embedded bitcode support
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+* If you combine `-fembed-bitcode` with `-flto`, no bitcode will be embedded into the binary
+* In some cases, llvm2cpg can't read debug information emitted by Xcode's version of Clang. Everything will work normally, but the debug information won't be taken into account.
+
+## Getting the CPG out of a Project
+
+Once you've obtain the bitcode for your project, you can get the CPG. The command required to do so depends on the method you used to obtain your bitcode.
+
+### `emit-llvm`
```
Comment (AlexDenisov) [inline using-ocular/tutorials/cpg-llvm.md]:
```suggestion
### `-emit-llvm`
```

File: using-ocular/tutorials/cpg-llvm.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+[2019-11-06 10:30:05.543] [llvm2cpg] [info] More details: /var/folders/pp/lt3pgm5971n1qw7pp2g_bmfr0000gn/T/llvm2cpg-d36a81.log
+[2019-11-06 10:30:05.543] [llvm2cpg] [info] Loading main.ll
+[2019-11-06 10:30:05.544] [llvm2cpg] [info] Emitting CPG 1/1
+[2019-11-06 10:30:05.544] [llvm2cpg] [info] Serializing CPG
+[2019-11-06 10:30:05.545] [llvm2cpg] [info] Saving CPG on disk
+[2019-11-06 10:30:05.546] [llvm2cpg] [info] CPG is successfully saved on disk: /tmp/sc-w4QviQTLdY/cpg.bin.zip
+[2019-11-06 10:30:05.546] [llvm2cpg] [info] Shutting down
+```
+
+Start Ocular by running `./ocular.sh`
```
Comment (arunmecheri) [inline using-ocular/tutorials/cpg-llvm.md]: shouldn't it be "sl ocular" and not invoking the shell script directly?

**Pull request merged**